vowpalwabbit/parse_args.cc

```diff
@@ -204,8 +203,8 @@ void parse_dictionary_argument(vw& all, std::string str)
                                                          << ", opening failed");
   }
 
-  feature_dict* map = &calloc_or_throw<feature_dict>();
-  map->init(1023, nullptr, substring_equal);
+  std::shared_ptr<feature_dict> map = std::make_shared<feature_dict>();
+  map->max_load_factor(0.25);
```

Comment:
0.25 seems very sparse. Is this intended?

(We should consider adding a comment explaining the choice if we know the reasoning behind it)